pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -485,286 +521,90 @@ async def parse_distribution_tree(self):
-                    dc.content.is_modular = True
-                    for dc_modulemd in self.data.nevra_to_module[dc.content.nevra]:
-                        dc.extra_data["modulemd_relation"].append(dc_modulemd)
-                        dc_modulemd.extra_data["package_relation"].append(dc)
-
-                if dc.content.name in self.data.pkgname_to_groups.keys():
-                    for dc_group in self.data.pkgname_to_groups[dc.content.name]:
-                        dc.extra_data["group_relations"].append(dc_group)
-                        dc_group.extra_data["related_packages"].append(dc)
-
-                packages_pb.increment()
-                await self.put(dc)
-
-    async def _parse_advisories(self, updates):
-        progress_data = {
-            "message": "Parsed Advisories",
-            "code": "sync.parsing.advisories",
-            "total": len(updates),
-        }
-        with ProgressReport(**progress_data) as advisories_pb:
-            for update in updates:
-                update_record = UpdateRecord(**UpdateRecord.createrepo_to_dict(update))
-                update_record.digest = hash_update_record(update)
-                future_relations = {"collections": defaultdict(list), "references": []}
-
-                for collection in update.collections:
-                    coll_dict = UpdateCollection.createrepo_to_dict(collection)
-                    coll = UpdateCollection(**coll_dict)
-
-                    for package in collection.packages:
-                        pkg_dict = UpdateCollectionPackage.createrepo_to_dict(package)
-                        pkg = UpdateCollectionPackage(**pkg_dict)
-                        future_relations["collections"][coll].append(pkg)
-
-                for reference in update.references:
-                    reference_dict = UpdateReference.createrepo_to_dict(reference)
-                    ref = UpdateReference(**reference_dict)
-                    future_relations["references"].append(ref)
-
-                advisories_pb.increment()
-                dc = DeclarativeContent(content=update_record)
-                dc.extra_data = future_relations
-                await self.put(dc)
-
-
-class FirstStageData:
-    """A data class that holds data required for synchronization.
-
-    The stored data are passed between multiple worker classes and are altered during
-    the synchronization process.
-    """
-
-    def __init__(self):
-        """Store shared data which are going to be used by parsers."""
-        self.metadata_pb = None
-
-        self.nevra_to_module = defaultdict(dict)
-        self.pkgname_to_groups = defaultdict(list)
-
-        self.package_repodata_urls = {}
-        self.repomd = None
-        self.remote_url = None
-        self.comps_url = None
-        self.modules_url = None
-        self.updateinfo_url = None
-        self.modulemd_list = []
-        self.dc_groups = []
-
-
-class RepositoryMetadataParser:
```

Comment:
I didn't see any justification for having classes for these, functions work just as well.  Logic was moved to `parse_repository_metadata`